Add after_response callback

sos4nt opened the pull request:
Billomat's API comes with [rate limiting](https://www.billomat.com/api/grundlagen/zugriffsbegrenzung/) which is currently not handled by the Billomat gem. And since the current rate limit values are included in the response headers which are not exposed by the Billomat gem, there's no easy way to account for a proper rate limit handling either.

Therefore, I've added a simple callback mechanism that's invoked upon API requests (or API responses to be more precise). It can be used like this:

```ruby
Billomat::Gateway.on_api_call = lambda do |response|
  remaining = response.headers[:x_rate_limit_remaining].to_i
  reset     = response.headers[:x_rate_limit_reset].to_i

  if limit < 10
    # stop processing until "reset"
  end
end
```

Since `on_api_call` can be anything that responds to `call`, you could also have a custom class handle the callback, e.g.

```ruby
Billomat::Gateway.on_api_call = MyRateLimitHandler

class MyRateLimitHandler
  def self.call(response)
    # ...
  end
end
```

or some method:

```ruby
Billomat::Gateway.on_api_call = MyRateLimitHandler.method(:billomat_api_call)

class MyRateLimitHandler
  def self.billomat_api_call(response)
    # ...
  end
end
```

Jack12816 reviewed (changes requested):
Thank you for your contribution! I like this solution. The only thing I would like to change is the hook name to `after_response`. Than it's more clear when it fires from my point of view. And maybe we can add a `before_response` hook in the future too, to manipulate the preconfigured request. (Not planned by us, but I can think of something would be handy, like the after-variant from you)

Could you please add some documentation example at the README? And please add a line to the change log under `next` which annotates the introduction of this feature. (Add `(#22)` as line suffix)

Jack12816 commented:
I've checked the gem sources again, and I would argue to move this from `Billomat::Gateway.on_api_call` / `Billomat::Gateway.after_response` to the gem configuration.

```ruby
Billomat.configure do |config|
  config.after_response = ->(resp) { .. }
end
```

Could you refactor this? :)

sos4nt commented:
I've renamed `on_api_call` to `after_response`, moved it to `Configuration` (as an instance method now), added a change log entry and included a basic usage example to the README. I've also rebased the branch against master for a clean history.

Regarding the _before_ hook: I didn't add it yet, but maybe `before_request` (rather than `before_response`) would be a better choice if the _request_ object is being passed along for adjustments / inspection. This way, each callback would receive the object corresponding to its name.

Jack12816 commented:
Great! Thank you very much for the contribution!
